src/IIIFPresentation/DLCS/API/DlcsApiClient.cs

```diff
@@ -146,6 +148,39 @@ await CallDlcsApiFor<HydraCollection<JObject>>(HttpMethod.Post, endpoint, hydraI
         return results;
     }
     
+    public async Task<IList<JObject>> GetCustomerImages(int customerId, string manifestId,
+        CancellationToken cancellationToken = default)
+    {
+        logger.LogTrace("Requesting images for customer {CustomerId} for the manifest {ManifestId}", customerId,
+            manifestId);
+
+        var page = 1;
+        var lastPage = false;
+        var results = new List<JObject>();
+
+        while (!lastPage)
+        {
+            var endpoint = $"/customers/{customerId}/allImages?q={{\"manifests\": [\"{manifestId}\"]}}&pageSize={settings.MaxImageListSize}&page={page}";
+            var result =
+                await CallDlcsApiFor<HydraCollection<JObject>>(HttpMethod.Get, endpoint, null, cancellationToken);
+            
+            if (result?.Members is {Length: > 0} pageResults)
+                results.AddRange(pageResults);
+
+            // use the result page size as this can be overwritten by the DLCS if we ask for more than allowed
+            if (result != null && result.PageSize * page < result.TotalItems)
```

Comment:
Not for now but I wonder if we would benefit from doing parallel requests for large batches? e.g. request page 1, see there are 10 pages and request in batches of 3 or something.